Veeam/PRTG-VeeamBRStats.ps1

```diff
@@ -145,249 +157,270 @@ if ($scaleouts) {
+    $Count = $failsSessionsEP.Count
+    Write-Output "<result>"
+                 "  <channel>Failes-Endpoints</channel>"
+                 "  <value>$Count</value>"
+                 "  <showChart>1</showChart>"
+                 "  <showTable>1</showTable>"
+                 "  <LimitMaxError>0</LimitMaxError>"
+                 "  <LimitMode>1</LimitMode>"
+                 "</result>"
+    $Count = $runningSessionsEP.Count
+    Write-Output "<result>"
+                 "  <channel>Running-Endpoints</channel>"
+                 "  <value>$Count</value>"
+                 "  <showChart>1</showChart>"
+                 "  <showTable>1</showTable>"
+                 "</result>"
+}
+#endregion:
+
+#region: Repository
+$RepoReport = $repoList | Get-vPCRepoInfo | Select-Object   @{Name="Repository Name"; Expression = {$_.Target}},
+                                                            @{Name="Host"; Expression = {$_.RepoHost}},
+                                                            @{Name="Path"; Expression = {$_.Storepath}},
+                                                            @{Name="Free (GB)"; Expression = {$_.StorageFree}},
+                                                            @{Name="Total (GB)"; Expression = {$_.StorageTotal}},
+                                                            @{Name="Free (%)"; Expression = {$_.FreePercentage}},
+                                                            @{Name="Status"; Expression = {
+                                                            If ($_.FreePercentage -lt $repoCritical) {"Critical"}
+                                                            ElseIf ($_.FreePercentage -lt $repoWarn) {"Warning"}
+                                                            ElseIf ($_.FreePercentage -eq "Unknown") {"Unknown"}
+                                                            Else {"OK"}}} | `
+                                                            Sort-Object "Repository Name"
 
 foreach ($Repo in $RepoReport){
 $Name = "REPO - " + $Repo."Repository Name"
 $Free = $Repo."Free (%)"
 Write-Output "<result>"
-               "<channel>$Name</channel>"
-               "<value>$Free</value>"
-               "<unit>Percent</unit>"
-               "<showChart>1</showChart>"
-               "<showTable>1</showTable>"
-               "<LimitMinWarning>$repoWarn</LimitMinWarning>"
-               "<LimitMinError>$repoCritical</LimitMinError>"
-               "<LimitMode>1</LimitMode>"
-               "</result>"
-	}
-Write-Output "</prtg>"
+             "  <channel>$Name</channel>"
+             "  <value>$Free</value>"
+             "  <unit>Percent</unit>"
+             "  <showChart>1</showChart>"
+             "  <showTable>1</showTable>"
+             "  <LimitMinWarning>$repoWarn</LimitMinWarning>"
+             "  <LimitMinError>$repoCritical</LimitMinError>"
+             "  <LimitMode>1</LimitMode>"
+             "</result>"
+}
 #endregion
 
-#region: Debug
```

Comment:
Why should the Debug Output be removed?

Reply:
In my eyes the debug output is not necessary, since PRTG has the possibility to save the output of the XML sensor to disk. It is sufficient to look at the XML in case of errors. This is why, the XML output has been reformatted to be better readable.

One can argue that the tabular output is more readable / compact. But it finally is not parsed by PRTG and other error sources (e.g. incorrect XML formatting) may be the source of the problem. So i left it away.

Looking forward to your opinion.